sql/hive/src/test/scala/org/apache/spark/sql/sources/hadoopFsRelationSuites.scala

```diff
@@ -76,6 +76,12 @@ abstract class HadoopFsRelationTest extends QueryTest with SQLTestUtils {
       df.filter('a > 1 && 'p1 < 2).select('b, 'p1),
       for (i <- 2 to 3; _ <- Seq("foo", "bar")) yield Row(s"val_$i", 1))
```

Comment:
@JoshRosen Found that I made a mistake in this `checkAnswer` test, which should have caught this bug... Values of `p1` should be either `"foo"` or `"bar"`, but I put a `1` there. The correct version of this test should be:

```scala
    // Simple projection and partition pruning
    checkAnswer(
      df.filter('a > 1 && 'p1 < 2).select('b, 'p1),
      for (i <- 2 to 3; j <- Seq("foo", "bar")) yield Row(s"val_$i", j))
```